docs/development_suite/set-up-infrastructure/enhanced-project-workflow.md

```diff
@@ -0,0 +1,136 @@
+
+The GitOps support feature can be activated in different ways, depending on your needs. In order to access the feature, please request the support of a Mia-Platform referent.
+
+You can activate the `ENABLE_CONF_GENERATION_ON_DEPLOY` feature toggle on a new project or Company to enable the enhanced project workflow. The feature toggle activates the new versioning system and the generation of Kubernetes configurations at the time of deployment. You can enable it for a single project or for an entire Company, which means all projects in the Company will adopt the new workflow.
+
+If you already have a Company with some projects, and you do not wish to migrate them to the new approach just yet, you can activate the `ENABLE_CREATE_PROJECT_WITH_SAVE_CONFIG_ON_DEPLOY` feature toggle on the Company. This feature toggle makes sure that all new projects in the Company will be created with the new workflow, while leaving the existing ones untouched. 
+
+:::note
+If both feature toggles are enabled, `ENABLE_CONF_GENERATION_ON_DEPLOY` will prevail.
+:::
+
+:::info
+In order to set up a new project, follow the [project creation guide](/development_suite/set-up-infrastructure/create-project.mdx). To create a new Company or to activate feature toggles, please open a Service Request.
+:::
+
+## Manual adjustments on newly created projects
+
+After creating a new project and activating the feature toggles, there are some important manual adjustments necessary to make sure the project works correctly.
+
+First of all, the project model **must have set a default revision**: ask your Mia-Platform referent to check it, and in case it is missing, to add it manually. You can choose a custom name for your default revision, e.g. `main`.
+
+If you want the project to use a **pull-based deployment strategy**,<!-- TODO: (LINK) --> its configuration must contain the `strategy` property set to `pull` (the value for push-based deployments is `push`). Based on the selected strategy, the `runnerTool` may take different values. 
+<!-- TODO: more details? -->
+
+Next, the project **Git repository structure** should contain:
+
+- A `configuration` directory, with all your global customizations to be applied to all environments, and a `kustomization.yaml` manifest if you are using Kustomize.
+- An `overlays` directory, containing one directory for each environment, inside which all your environment-specific patch files are stored.
+- An `environments` directory, automatically generated by the Console, containing one directory for each environment, inside which all the necessary configurations are generated and stored after each push.
+
+There are some **points of attention** and possible modifications needed to make sure everything works properly on the Git repository:
+
+- If there are any configuration files in the repository root (e.g. `api-console-config.json`), you can delete them, since they refer to the old management strategy of configuration files.
```

Comment:
This is about "migrating" an existent project; I think we have to distinguish the repo structure of newly created projects with already enabled the config generation at deploy time, from the ones that have been migrated